### ISSUE TITLE
Support generic parameters in "extract function" refactoring

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#2187]
Current "extract function" implementation doesn't add type parameters and their bounds for generic arguments into function signature.
It would be great to support it.

```rust
fn foo<T: Default>() -> T {
    <selection>T::default()</selection>
}
```

**Expected:**
```rust
fn foo<T: Default>() -> T {
    bar()
}

fn bar<T: Default>() -> T {
    T::default()
}
```

**Actual:**
```rust
fn foo<T: Default>() -> T {
    bar()
}

fn bar() -> T {
    T::default()
}
```

**Comment by t-kameyama** [comment]
I will work on this issue.

**Comment by Undin** [comment]
Implemented in #2197
